src/Uart/Uart.c

```diff
@@ -0,0 +1,441 @@
+/**@file
+ * This file is part of the Leon3 BSP for the Test Environment.
+ *
+ * @copyright 2022 N7 Space Sp. z o.o.
+ *
+ * Test Environment was developed under a programme of,
+ * and funded by, the European Space Agency (the "ESA").
+ *
+ *
+ * Licensed under the ESA Public License (ESA-PL) Permissive,
+ * Version 2.3 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://essr.esa.int/license/list
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/// \brief Uart hardware driver implementation.
+
+#include <stdbool.h>
+#include "Uart.h"
+#include "UartRegisters.h"
+#include "ByteFifo.h"
+#include "SystemConfig.h"
+#include <rtems.h>
+
+#define GPTIMER_ADDRESS_BASE 0x80000300U
+
+#ifdef RTEMS_SIS
+#define UART_INTERFACES_MAX 2
+#else
+#define UART_INTERFACES_MAX 6
+#endif
+
+static Uart_ErrorFlags errorFlags[UART_INTERFACES_MAX];
+
+static uint32_t sentBytesCounter[UART_INTERFACES_MAX];
+
+static rtems_interrupt_entry uartIrqEntry[UART_INTERFACES_MAX];
```

Comment:
I missed this one the last time... Why there are separate static arrays containing data for each Uart... when we could store the relevant data in the Uart struct?

Reply:
We can, of course; this solution was selected only to keep API mostly identical with SAMV71.